processing/src/main/java/org/apache/druid/query/filter/RangeFilter.java

```diff
@@ -0,0 +1,1025 @@
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return ((lowerComparing > 0)) && (upperComparing > 0);
+          };
+        } else if (lowerStrict) {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return (lowerComparing > 0) && (upperComparing >= 0);
+          };
+        } else if (upperStrict) {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return (lowerComparing >= 0) && (upperComparing > 0);
+          };
+        } else {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return (lowerComparing >= 0) && (upperComparing >= 0);
+          };
+        }
+      } else if (hasUpperBound()) {
+        if (upperStrict) {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return upperComparing > 0;
+          };
+        } else {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return upperComparing >= 0;
+          };
+        }
+      } else if (hasLowerBound()) {
+        if (lowerStrict) {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            return lowerComparing > 0;
+          };
+        } else {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            return lowerComparing >= 0;
+          };
+        }
+      } else {
+        return Predicates.notNull();
+      }
+    });
+  }
+
+  private Predicate<Object[]> makeArrayPredicate(@Nullable TypeSignature<ValueType> arrayType)
+  {
+    if (hasLowerBound() && hasUpperBound()) {
+      if (upperStrict && lowerStrict) {
```

Comment:
I ended up just making an actual enum, which I think is a bit nicer. I also rebranded 'strict' as 'open' since the enum verbage is around open and closed ranges